lua/nvim-tree/git/init.lua

```diff
@@ -143,29 +143,36 @@ function M.load_project_status(cwd)
     timeout = M.config.git.timeout,
   }
 
+  local git_directory = git_utils.get_git_directory(project_root)
```

Comment:
This is good, it's done only once.

Reply:
This should be moved inside `if M.config.filesystem_watchers.enable then` so that we don't unnecessarily execute this when watchers are disabled.